gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/tasks/CreateManifestTaskTest.groovy

```diff
@@ -1,10 +1,159 @@
 package com.palantir.gradle.dist.tasks
 
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.palantir.gradle.dist.GradleTestSpec
+import com.palantir.gradle.dist.RecommendedProductDependencies
+import com.palantir.gradle.dist.RecommendedProductDependency
+import nebula.test.dependencies.DependencyGraph
+import nebula.test.dependencies.GradleDependencyGenerator
 import org.gradle.api.Project
 import org.gradle.testfixtures.ProjectBuilder
-import spock.lang.Specification
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.TaskOutcome
 
-class CreateManifestTaskTest extends Specification {
+class CreateManifestTaskTest extends GradleTestSpec {
+
+    def jsonMapper = new ObjectMapper()
+            .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+            .setPropertyNamingStrategy(new KebabCaseStrategy())
+    def yamlMapper = new ObjectMapper(new YAMLFactory())
+            .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+            .setPropertyNamingStrategy(new KebabCaseStrategy())
+
+    File mavenRepo
+
+    def 'Fail on missing recommended product dependencies'() {
+        setup:
+        generateDependencies()
+        buildFile << '''
+            plugins {
+                id 'com.palantir.sls-java-service-distribution'
+            }
+
+            repositories {
+                maven {url "file:///{{mavenRepo}}"}
+            }
+
+            project.version = '1.0.0'
+
+            dependencies {
+                runtime 'a:a:1.0'
+            }
+
+            task testCreateManifest(type: com.palantir.gradle.dist.tasks.CreateManifestTask) {
+                serviceName = "serviceName"
+                serviceGroup = "serviceGroup"
+                productType = "service"
+                manifestExtensions = [:]
+                productDependencies = []
+                productDependenciesConfig = configurations.runtime
```

Comment:
runtime is created by the java plugin, asset bunldes don't have the java plugin applied